File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.py
def solution(array):
    s_array = set(array)
    
    ch = 0
    new_arr = [0]

    for i in s_array:
        if(ch < array.count(i)):
            ch = array.count(i)
            new_arr[0] = i
        elif(ch == array.count(i)):
            new_arr[0] = -1
    return new_arr[0]

File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.java
import java.util.HashMap;

class Solution {
    public int solution(int[] array) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int num: array){
            map.put(num, map.getOrDefault(num, 0) + 1);
        }
        
        int MaxCount = 0;
        int answer = 0;
        
        for(int key : map.keySet()){
            int count = map.get(key);
            if(MaxCount < count){
                answer = key;
                MaxCount = count;
            }
            else if(count == MaxCount){answer = -1;}
        }
        
        return answer;
    }
}